opentreemap/treemap/lib/user.py

```diff
@@ -112,20 +112,38 @@ def get_audits(logged_in_user, instance, query_vars, user, models,
     if exclude_pending:
         audits = audits.exclude(requires_auth=True, ref__isnull=True)
 
+    # Slicing the QuerySet uses a SQL Limit, which has proven to be quite slow.
+    # By relying on the fact the our list is ordered by primary key from newest
+    # to oldest, we can rely on the index on the primary key, which is faster.
+    if start_id is not None:
+        audits = audits.filter(pk__lt=start_id)
+
     total_count = audits.count() if should_count else 0
-    audits = audits[start_pos:end_pos]
+    audits = audits[:page_size]
+
+    # Coerce the queryset into a list so we can get the last item on the page
```

Comment:
I added a comment about this in https://github.com/OpenTreeMap/otm-core/pull/2904/commits/05859cc71145e7f874d79c2ea6d80b117b75e922

Basically - QuerySets support indexing in general, but not negative indexing - so if we want to get the last item in the query set (which corresponds to the last row in the edits table on the page), we need to coerce the QuerySet into a list, which does support negative indexing.